src/features/calendar/components/CalendarMonthView/index.tsx

```diff
@@ -1,5 +1,65 @@
-const CalendarMonthView = () => {
-  return <h2>Month</h2>;
+import { Box } from '@mui/system';
+import dayjs from 'dayjs';
+import isoWeek from 'dayjs/plugin/isoWeek';
+import React from 'react';
+
+import Day from './Day';
+import WeekNumber from './WeekNumber';
+import { getDayIndex, getDaysBeforeFirstDay, getWeekNumber } from './utils';
+
+dayjs.extend(isoWeek);
+
+type CalendarMonthViewProps = {
+  focusDate: Date;
+};
+
+const CalendarMonthView = ({ focusDate }: CalendarMonthViewProps) => {
+  const numberOfRows = 6;
+  const numberOfColumns = 7;
+
+  const firstDayOfMonth: Date = new Date(
+    focusDate.getFullYear(),
+    focusDate.getMonth(),
+    1
+  );
+  const firstDayOfCalendar: Date = dayjs(firstDayOfMonth)
+    .subtract(getDaysBeforeFirstDay(firstDayOfMonth), 'day')
+    .toDate();
+
+  return (
+    <Box
+      display="grid"
+      flexGrow="1"
+      gap="8px"
+      gridTemplateColumns={`auto repeat(${numberOfColumns}, 1fr)`}
+      gridTemplateRows={`repeat(${numberOfRows}, 1fr)`}
+    >
+      {[...Array(numberOfRows)].map((_, rowIndex) =>
+        [...Array(numberOfColumns + 1)].map((_, columnIndex) => (
+          <React.Fragment key={`${rowIndex * numberOfColumns + columnIndex}`}>
```

Comment:
I think row 37-39 could do with some clarification, it's maybe me not being used to this style of writing but I have a hard tim following what is going on here.

Reply:
Good point! Maybe we could create a utility. I also know some people use `lodash` for this, e.g. in the week view branch.